From c65c60a68c93946c693fcb05f5f045c4a58bbf26 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Wed, 1 Oct 2014 17:40:38 -0400
Subject: [ethernet] cxgb4: Allow T4/T5 firmware sizes up to 1MB

Message-id: <1412185238-11567-8-git-send-email-svemuri@redhat.com>
Patchwork-id: 96833
O-Subject: [RHEL7.1 PATCH BZ 1147683 7/7] cxgb4: Allow T4/T5 firmware sizes up to 1MB
Bugzilla: 1147683
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>

Upstream : https://git.kernel.org/cgit/linux/kernel/git/davem/net-next.git/commit/drivers/net/ethernet/chelsio?id=60d42bf6b0eec663ad21170b1ff6b14536993b47

Based on original work by Casey Leedom <leedom@chelsio.com>

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 60d42bf6b0eec663ad21170b1ff6b14536993b47)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.h b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.h
index 55c3a6a..10128d7 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.h
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.h
@@ -175,7 +175,7 @@ enum {
   * Location of firmware image in FLASH.
   */
  FLASH_FW_START_SEC = 8,
- FLASH_FW_NSECS = 8,
+ FLASH_FW_NSECS = 16,
  FLASH_FW_START = FLASH_START(FLASH_FW_START_SEC),
  FLASH_FW_MAX_SIZE = FLASH_MAX_SIZE(FLASH_FW_NSECS),
 
-- 
1.7.1